arbos/l1pricing/l1pricing.go

```diff
@@ -5,242 +5,358 @@ package l1pricing
+	if err := sto.SetUint64ByUint64(inertiaOffset, InitialInertia); err != nil {
 		return err
 	}
-	return sto.SetUint64ByUint64(lastL1BaseFeeUpdateTimeOffset, 0)
+	if err := sto.SetUint64ByUint64(perUnitRewardOffset, InitialPerUnitReward); err != nil {
+		return err
+	}
+	pricePerUnit := sto.OpenStorageBackedBigInt(pricePerUnitOffset)
+	return pricePerUnit.Set(big.NewInt(InitialPricePerUnitGwei * 1000000000))
 }
 
 func OpenL1PricingState(sto *storage.Storage) *L1PricingState {
 	return &L1PricingState{
 		sto,
-		sto.OpenStorageBackedAddress(defaultAggregatorAddressOffset),
-		sto.OpenStorageBackedBigInt(l1BaseFeeEstimateOffset),
-		sto.OpenStorageBackedUint64(l1BaseFeeEstimateInertiaOffset),
-		sto.OpenStorageBackedUint64(lastL1BaseFeeUpdateTimeOffset),
-		sto.OpenSubStorage(userSpecifiedAggregatorKey),
-		sto.OpenSubStorage(refuseDefaultAggregatorKey),
-		sto.OpenSubStorage(aggregatorFeeCollectorKey),
-		sto.OpenSubStorage(aggregatorCompressionRatioKey),
+		sto.OpenStorageBackedAddress(sequencerOffset),
+		sto.OpenStorageBackedAddress(paySequencerFeesToOffset),
+		sto.OpenStorageBackedAddress(payRewardsToOffset),
+		sto.OpenStorageBackedUint64(equilibrationTimeOffset),
+		sto.OpenStorageBackedUint64(inertiaOffset),
+		sto.OpenStorageBackedUint64(perUnitRewardOffset),
+		sto.OpenStorageBackedUint64(currentTimeOffset),
+		sto.OpenStorageBackedUint64(lastUpdateTimeOffset),
+		sto.OpenStorageBackedBigInt(availableFundsOffset),
+		sto.OpenStorageBackedBigInt(fundsDueToSequencerOffset),
+		sto.OpenStorageBackedBigInt(fundsDueForRewards),
+		sto.OpenStorageBackedUint64(unitsSinceOffset),
+		sto.OpenStorageBackedBigInt(pricePerUnitOffset),
 	}
 }
 
-func (ps *L1PricingState) DefaultAggregator() (common.Address, error) {
-	return ps.defaultAggregator.Get()
+func (ps *L1PricingState) Sequencer() (common.Address, error) {
+	return ps.sequencer.Get()
 }
 
-func (ps *L1PricingState) SetDefaultAggregator(val common.Address) error {
-	return ps.defaultAggregator.Set(val)
+func (ps *L1PricingState) SetSequencer(seq common.Address) error {
+	return ps.sequencer.Set(seq)
 }
 
-func (ps *L1PricingState) L1BaseFeeEstimateWei() (*big.Int, error) {
-	return ps.l1BaseFeeEstimate.Get()
+func (ps *L1PricingState) PaySequencerFeesTo() (common.Address, error) {
+	return ps.paySequencerFeesTo.Get()
 }
 
-func (ps *L1PricingState) SetL1BaseFeeEstimateWei(val *big.Int) error {
-	return ps.l1BaseFeeEstimate.Set(val)
+func (ps *L1PricingState) SetPaySequencerFeesTo(addr common.Address) error {
+	return ps.paySequencerFeesTo.Set(addr)
 }
 
-func (ps *L1PricingState) LastL1BaseFeeUpdateTime() (uint64, error) {
-	return ps.lastL1BaseFeeUpdateTime.Get()
+func (ps *L1PricingState) PayRewardsTo() (common.Address, error) {
+	return ps.payRewardsTo.Get()
 }
 
-func (ps *L1PricingState) SetLastL1BaseFeeUpdateTime(t uint64) error {
-	return ps.lastL1BaseFeeUpdateTime.Set(t)
+func (ps *L1PricingState) EquilibrationTime() (uint64, error) {
```

Comment:
There's no setter for this. We should probably add an `ArbOwner` method

Reply:
Done